internal/cloud/realm/secrets.go

```diff
@@ -63,3 +65,22 @@ func (c *client) CreateSecret(groupID, appID, name, value string) (Secret, error
 	}
 	return secret, nil
 }
+
+func (c *client) DeleteSecret(groupID, appID, secretID string) error {
+	res, resErr := c.do(
+		http.MethodDelete,
+		fmt.Sprintf(secretPathPattern, groupID, appID, secretID),
+		api.RequestOptions{},
+	)
+
+	if resErr != nil {
+		return resErr
+	}
+
+	if res.StatusCode != http.StatusNoContent {
+		defer res.Body.Close()
```

Comment:
and you won't need the `defer` here, since we never actually attempt to read the `res.Body` anywhere here